bin/plugin/commands/test-plugins.js

```diff
@@ -0,0 +1,503 @@
+ */
+
+exports.options = [
+	{
+		argname: '-s, --sitetype <siteType>',
+		description: 'Whether to test "single" (default) or "multi" site.',
+	},
+];
+
+// Switch for flagging start of wp-env so it is not started multiple times.
+let isWpEnvStarted = false;
+
+/**
+ * Command for testing all built standalone plugins.
+ *
+ * @param {WPTestPluginsCommandOptions} opt Command options.
+ */
+exports.handler = async ( opt ) => {
+	doRunStandalonePluginTests( {
+		pluginsJsonFile: './plugins.json', // Path to plugins.json file.
+		siteType: opt.sitetype || 'single', // Site type.
+		pluginTestAssets: './plugin-tests', // plugin test assets.
+		builtPluginsDir: './build/', // Built plugins directory.
+		wpEnvFile: './plugin-tests/.wp-env.json', // Base .wp-env.json file for testing plugins.
+		wpEnvDestinationFile: './.wp-env.override.json', // Destination .wp-env.override.json file at root level.
+		wpEnvPluginsRegexPattern: '"plugins": \\[(.*)\\],', // Regex to match plugins string in .wp-env.json.
+		performancePluginSlug: 'performance', // The plugin slug of the main WPP plugin.
+	} );
+};
+
+/**
+ * Handles replacement of plugins array in .wp-env.json file.
+ * Split into separate function in order to easily re-run tests with WPP plugin active.
+ *
+ * @param {WPDoReplaceWpEnvContent} settings Plugin test settings.
+ */
+function doReplaceWpEnvContent( settings ) {
+	// Regex object to match wp-env plugins string.
+	const wpEnvPluginsRegex = new RegExp( settings.wpEnvPluginsRegexPattern, 'gm' );
+
+	let wpEnvPluginsRegexReplacement = '';
+
+	// Amend wp-env overrides file to reference built plugins only.
+	// Buffer .wp-env.json content var.
+	let wpEnvFileContent = '';
+
+	try {
+		wpEnvFileContent = fs.readFileSync( settings.wpEnvFile, 'utf-8' );
+	} catch ( e ) {
+		log( formats.error( `Error reading file "${ settings.wpEnvFile }": "${ e }"` ) );
+
+		// Return with exit code 1 to trigger a failure in the test pipeline.
+		process.exit( 1 );
+	}
+
+	// If the contents of the file were incorrectly read or exception was not captured and value is blank, abort.
+	if ( '' === wpEnvFileContent ) {
+		log(
+			formats.error(
+				`File content for "${ settings.wpEnvFile }" is empty, aborting.`
+			)
+		);
+
+		// Return with exit code 1 to trigger a failure in the test pipeline.
+		process.exit( 1 );
+	}
+
+	// If we do not have a match on the wp-env enabled plugins regex, abort.
+	if ( ! wpEnvPluginsRegex.test( wpEnvFileContent ) ) {
+		log(
+			formats.error(
+				`Unable to find plugins property/key in WP Env config file: "${ settings.wpEnvFile }". Please ensure that it is present and try again.`
+			)
+		);
+
+		// Return with exit code 1 to trigger a failure in the test pipeline.
+		process.exit( 1 );
+	}
+
+	// Copy the newly modified wp-env overrides file to the root level if it does not exist.
+	if ( ! fs.pathExistsSync( settings.wpEnvDestinationFile ) ) {
+		try {
+			fs.copySync( settings.wpEnvFile, settings.wpEnvDestinationFile, {
```

Comment:
At the very end of the entire process, we need to make sure we delete the `wpEnvDestinationFile` again. While it's in `.gitignore` which is a good idea, we need to still delete it since otherwise we'll "pollute" the development directory in a way that using `wp-env` will no longer result in the expected Performance Lab plugin environment but rather the one from the overridden configuration.

Reply:
@felixarntz addressed in a recent commit, thank you.